04-advanced-topics/network-analysis_rnaseq_01_wgcna.Rmd

```diff
@@ -420,13 +442,271 @@ readr::write_rds(bwnet,
+  col_annot <- ComplexHeatmap::HeatmapAnnotation(
+    # Supply treatment labels
+    refinebio_treatment = col_annot_df$refinebio_treatment,
+    # Add annotation barplot
+    module_eigengene = ComplexHeatmap::anno_barplot(dplyr::select(col_annot_df, module_name)),
+    # Pick colors for each experimental group in refinebio_treatment
+    col = list(refinebio_treatment = c("post-adt" = "#f1a340", "pre-adt" = "#998ec3"))
+  )
+
+  # Get a vector of the Ensembl gene ides that correspond to this module
+  module_genes <- gene_module_key_df %>%
+    dplyr::filter(module == module_name) %>%
+    dplyr::pull(gene)
+
+  # Set up the gene expression data frame
+  mod_mat <- expression_mat %>%
+    t() %>%
+    as.data.frame() %>%
+    # Only keep genes from this module
+    dplyr::filter(rownames(.) %in% module_genes) %>%
+    # Order the samples to match col_annot_df
+    dplyr::select(rownames(col_annot_df)) %>%
+    # Data needs to be a matrix
+    as.matrix()
+
+  mod_mat <- mod_mat %>%
+    # Scale can work on matrices, but it does it by column so we will need to
+    # transpose first
+    t() %>%
+    scale() %>%
+    # And now we need to transpose back
+    t()
+
+  # Create a color function based on standardized scale
+  color_func <- circlize::colorRamp2(
+    c(-2, 0, 2),
+    c("#67a9cf", "#f7f7f7", "#ef8a62")
+  )
+
+  # Plot on a heatmap
+  heatmap <- ComplexHeatmap::Heatmap(mod_mat,
+    name = module_name,
+    # Supply color function
+    col = color_func,
+    # Supply column annotation
+    bottom_annotation = col_annot,
+    # We don't want to cluster samples
+    cluster_columns = FALSE,
+    # We don't need to show sample or gene labels
+    show_row_names = FALSE,
+    show_column_names = FALSE
+  )
+
+  # Return heatmap
+  return(heatmap)
+}
+```
+
+## Make module heatmaps
+
+Let's try out the custom heatmap function with module 52 (our most differentially expressed module).
+
+```{r}
+mod_52_heatmap <- make_module_heatmap(module_name = "ME52")
+
+# Print out the plot
+mod_52_heatmap
+```
+From the barplot portion of our plot, we can see `post-adt` samples have higher values for this eigengene for module 52.
+In the heatmap portion, we can see how the individual genes that make up module 52 have more extreme values (very high or very low) in the `post-adt` samples. 
+
+We can save this plot to PDF.
+
+```{r}
+pdf(file.path("results", "SRP133573_module_52_heatmap.pdf"))
+mod_52_heatmap
+dev.off()
+```
+
+For comparison, let's try out the custom heatmap function with a different, _not_ differentially expressed module. 
+
+```{r}
+mod_10_heatmap <- make_module_heatmap(module_name = "ME10")
+
+# Print out the plot
+mod_10_heatmap
+```
+In this non-significant module's heatmap, there's not a particularly strong pattern between pre and post ADT samples.
+In general expression is pretty low for all the genes that make up module 10 but there's a few samples and some genes that show higher expression. 
```

Comment:
Adding a bit more detail and clarification that "low" doesn't necessarily mean low, since these are normalized values.
```suggestion
In general the expression of genes in module 10 does not vary much between groups, staying near the overall mean. 
There are a few samples and some genes that show higher expression, but it is not surprising this does not results in a significant overall difference between the groups.
```